frontend/src/views/TE/Upload.vue

```diff
@@ -198,18 +273,17 @@ export default class Upload extends Vue {
     let files = Array.isArray(selectedFiles)
       ? selectedFiles
       : Object.values(selectedFiles);
-    files.forEach(file => {
-      this.files.set(file, {});
-      try {
-        validateFlacFile(file);
-        this.handleFile(file, timestamp);
-      } catch (e) {
-        this.emitFileError(file, e.message);
-      }
-    });
+    for (const file of files) {
+      this.updateFileFields(file, {});
+      this.validateUpload(file)
+        .then(() => this.handleFile(file, timestamp))
+        .catch(e =>
+          this.emitFileError(file, e instanceof FileUploadError ? e : e.message)
+        );
```

Comment:
Is there any particular reason why you changed to chaining `then` and `catch` instead of `try ... catch`? Perhaps because the validation became async... Can't we just `await` it?

Reply:
Using `await` would prevent the next file validation until the one before it is validated.

Reply:
I see, good point.